src/test/scala/GameServiceTest.scala

```diff
@@ -1,79 +1,59 @@
+import org.scalatest.OneInstancePerTest
 import org.scalatest.funspec.AnyFunSpec
 import org.scalamock.scalatest.MockFactory
-import service.GameServiceImpl
-import service.GameService
+import org.scalacheck.Gen
+import io.buildo.enumero.CaseEnumSerialization
+import service.{CpuMoveStrategy, GameEngine, GameService, GameServiceImpl}
 import persistance.GameRepository
 import command.Move
 import command.Move._
 import model.GameResult._
+import model.FinalResult
 import persistance.GameStatusEntity
-import service.CpuMoveStrategy
+import model.GameResult
 
-class GameServiceTest extends AnyFunSpec with MockFactory with BeforeAndAfterEach {
+class GameServiceTest extends AnyFunSpec with MockFactory with OneInstancePerTest {
 
-  var mockgameRepository: GameRepository = _
-  var mockCpuMoveStrategy: CpuMoveStrategy = _
-  var gameService: GameServiceImpl = _
+  var mockgameRepository: GameRepository = mock[GameRepository]
+  var mockCpuMoveStrategy: CpuMoveStrategy = mock[CpuMoveStrategy]
+  var mockGameEngine: GameEngine = mock[GameEngine]
 
-  override def beforeEach() {
-    mockgameRepository = mock[GameRepository]
-    mockCpuMoveStrategy = mock[CpuMoveStrategy]
-    gameService = new GameServiceImpl(mockgameRepository, mockCpuMoveStrategy)
-  }
-
-  describe("Test play") {
-
-    it("user lose") {
-
-      (mockCpuMoveStrategy.provideCPUMove _).expects().returns(Paper)
-      (mockgameRepository.insertGameResult _).expects(Rock, Paper, Lose).once()
-
-      val actualResult = gameService.play(Rock)
-
-      assert(actualResult.gameResult == Lose)
-      assert(actualResult.computerMove == Paper)
-      assert(actualResult.userMove == Rock)
-    }
-
-    it("user win") {
-      (mockCpuMoveStrategy.provideCPUMove _).expects().returns(Scissors)
-      (mockgameRepository.insertGameResult _).expects(Rock, Scissors, Win).once()
+  val gameResultsGen: Gen[GameStatusEntity] = for {
+    userMove <- Gen.oneOf(CaseEnumSerialization.apply[Move].values)
+    cpuMove <- Gen.oneOf(CaseEnumSerialization.apply[Move].values)
+    result <- Gen.oneOf(CaseEnumSerialization.apply[GameResult].values)
+  } yield GameStatusEntity(userMove, cpuMove, result)
 
-      val actualResult = gameService.play(Rock)
+  val genGameResultsList = (num: Int) => Gen.containerOfN[List, GameStatusEntity](num, gameResultsGen)
 
-      assert(actualResult.gameResult == Win)
-      assert(actualResult.computerMove == Scissors)
-      assert(actualResult.userMove == Rock)
-    }
+  var gameService: GameServiceImpl = new GameServiceImpl(mockgameRepository, mockCpuMoveStrategy, mockGameEngine)
 
-    it("draw case") {
-      (mockCpuMoveStrategy.provideCPUMove _).expects().returns(Scissors)
-      (mockgameRepository.insertGameResult _).expects(Scissors, Scissors, Draw).once()
+  it("Test play behaviours") {
+    (mockCpuMoveStrategy.provideCPUMove _).expects().returns(Paper)
+    (mockGameEngine.evaluateWinner _).expects(Rock, Paper).returns(Lose)
```

Comment:
maybe you can use the real implementation here? Do we really need to mock it?

Also, maybe we can make the test a little bit more interesting using scalacheck? You could test that:

```
for every (cpuMove, userMove)
   given that provideCPUMove returns userMove
      gameService.play(userMove) returns (userMove, cpuMove, gameEngine.evaluateWinner(userMove, cpuMove))
```

what do you think? (cc @gabro)